tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_create_one_docker_repo(self, name):
+        """@Test: Create one Docker-type repository
+
+        @Assert: A repository is created with a Docker image.
+
+        @Feature: Docker
+
+        """
+        upstream_name = u'busybox'
+        content_type = u'docker'
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+
+        repository = _create_repository(prod_id, name, None)
+        repo_id = repository['id']
+        real_attrs = entities.Repository(id=repo_id).read_json()
+        self.assertEqual(real_attrs['name'], name)
+        self.assertEqual(real_attrs['docker_upstream_name'], upstream_name)
+        self.assertEqual(real_attrs['content_type'], content_type)
+
+    @run_only_on('sat')
+    def test_create_multiple_docker_repo(self):
+        """@Test: Create multiple Docker-type repositories
+
+        @Assert: Multiple docker repositories are created with a Docker image
+        and they all belong to the same product.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+
+        for _ in range(0, randint(1, 5)):
+            repository = _create_repository(prod_id, None, None)
+            repo_id = repository['id']
+            prod_attrs = entities.Product(id=prod_id).read_json()
+            self.assertIn(repo_id, [
+                repo['id'] for repo in prod_attrs['repositories']])
+
+    @run_only_on('sat')
+    def test_create_multiple_docker_repo_multiple_products(self):
+        """@Test: Create multiple Docker-type repositories on multiple products.
+
+        @Assert: Multiple docker repositories are created with a Docker image
+        and they all belong to their respective products.
+
+        @Feature: Docker
+
+        """
+        for _ in range(0, randint(1, 5)):
+            prod_id = entities.Product(organization=self.org_id).create()['id']
+            for _ in range(0, randint(1, 3)):
+                repository = _create_repository(prod_id, None, None)
+                repo_id = repository['id']
+                prod_attrs = entities.Product(id=prod_id).read_json()
+                self.assertIn(repo_id, [
+                    repo['id'] for repo in prod_attrs['repositories']])
+
+    @run_only_on('sat')
+    def test_sync_docker_repo(self):
+        """@Test: Create and sync a Docker-type repository
+
+        @Assert: A repository is created with a Docker repository
+        and it is synchronized.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+        repository = _create_repository(prod_id, None, None)
+        repo_id = repository['id']
+
+        task_result = entities.Repository(id=repo_id).sync()['result']
+        self.assertEqual(
+            u'success',
+            task_result)
```

Comment:
This may produce better error messages:

    task = entities.Repository(id=repo_id).sync()
    self.assertEqual('success', task['result'], task)